test/unit/backends/test_abstract.rb

```diff
@@ -112,14 +112,14 @@ def test_within_home
       end
 
       def test_background_logs_deprecation_warnings
-        deprecation_out = ''
+        deprecation_out = []
```

Comment:
Changing this to an array slightly alters the implied API contract of the deprecation logger, which expects an output stream. A more lightweight change could be:

```suggestion
        deprecation_out = +''
```

This would create a mutable string instead of a frozen one.

Alternatively, to be more explicit about passing a stream-like object, we could do this:

```suggestion
        deprecation_out = StringIO.new
```

I think both would be preferable to passing an array.

Reply:
Hmmm looks like +'' isnt a thing in ruby 2.0 and StringIO doesnt support << or lines. How would you like to proceed here? 

<img width="368" alt="image" src="https://github.com/user-attachments/assets/c8bc8e51-b548-4c95-b9b7-557a0f80c36f">

Reply:
Actually nevermind, think i figured it.